doxygen/modules/template/module-template.config

```diff
@@ -0,0 +1,322 @@
+# Doxyfile 1.8.11
+
+#---------------------------------------------------------------------------
+# Project related configuration options
+#---------------------------------------------------------------------------
+DOXYFILE_ENCODING      = UTF-8
+PROJECT_NAME           = "AWS SDK for C++"
+PROJECT_NUMBER         = "1.10.12"
```

Comment:
Shouldn't this one be a placeholder?

Reply:
Its replaced with a `sed` statement, and was previously i.e.

```
'sed -i "s/PROJECT_NUMBER .*/PROJECT_NUMBER         = $VERSION_NUM/" ./doxygen/modules/static/root.config',
```


 at runtime, so it doesnt matter what is there. previously, actually currently until we do this, [it is also hard coded](https://github.com/aws/aws-sdk-cpp/blob/main/doxygen/doxygen.config#L8) and replaced by sed at runtime. its actually set to `1.3.52` previously and replaced. in reality anything can be there.